File: design-patterns/go/decorator/data_export/data_export_decorator.go
// data_export_decorator.go

package main

type DataExportDecorator struct {
	dataExport DataExport
}

func NewDataExportDecorator(dataExport DataExport) (dataExportDecorator *DataExportDecorator) {
	dataExportDecorator = &DataExportDecorator{}
	dataExportDecorator.dataExport = dataExport
	return
}

func (dataExportDecorator *DataExportDecorator) ProcessData() {
	dataExportDecorator.dataExport.ProcessData()
}

File: design-patterns/ts/decorator/data-export/data-export-decorator.ts
// data-export-decorator.ts

import DataExport from "./data-export";


class DataExportDecorator implements DataExport {
    protected dataExporter: DataExport;

    constructor(dataExporter: DataExport) {
        this.dataExporter = dataExporter;
    }

    public processData() {
        this.dataExporter.processData();
    }
}

export default DataExportDecorator;